src/ring_theory/integral_domain.lean

```diff
@@ -30,25 +30,31 @@ section
 open finset polynomial function
 open_locale big_operators nat
 
+section cancel_monoid_with_zero
+-- There doesn't seem to be a better home for these right now
+variables {M : Type*} [cancel_monoid_with_zero M] [fintype M]
+
+lemma mul_right_bijective_of_fintype₀ (a : M) (ha : a ≠ 0) : bijective (λ b, a * b) :=
+fintype.injective_iff_bijective.1 $ mul_right_injective₀ ha
+
+lemma mul_left_bijective_of_fintype₀ (a : M) (ha : a ≠ 0) : bijective (λ b, b * a) :=
+fintype.injective_iff_bijective.1 $ mul_left_injective₀ ha
+
+end cancel_monoid_with_zero
+
 variables {R : Type*} {G : Type*}
 
 section ring
 
 variables [ring R] [is_domain R] [fintype R]
 
-lemma mul_right_bijective₀ (a : R) (ha : a ≠ 0) : bijective (λ b, a * b) :=
-fintype.injective_iff_bijective.1 $ mul_right_injective₀ ha
-
-lemma mul_left_bijective₀ (a : R) (ha : a ≠ 0) : bijective (λ b, b * a) :=
-fintype.injective_iff_bijective.1 $ mul_left_injective₀ ha
-
 /-- Every finite domain is a division ring.
 
 TODO: Prove Wedderburn's little theorem,
 which shows a finite domain is in fact commutative, hence a field. -/
 def division_ring_of_is_domain (R : Type*) [ring R] [is_domain R] [decidable_eq R] [fintype R] :
```

Comment:
Should there be a `group_with_zero_of_fintype`?

Reply:
What do you mean? A def saying that every `fintype` `cancel_monoid_with_zero` is a `group_with_zero`?

Reply:
Yes exactly; you should just be able to reuse the proof below, then implement `division_ring_of_is_domain` in terms of group_with_zero_of_fintype

Reply:
Ok, done, any ideas where this def can go now? It really feels out of place at this point but I don't know a good file for finite groups with zero (iirc the group with zero files don't have fintype imported right now)

Reply:
I think moving it out of this file is probably out of scope for this PR :)